cmd/cluster-init/main.go

```diff
@@ -46,12 +69,42 @@ func validateOptions(o options) []error {
 	if o.releaseRepo == "" {
 		//If the release repo is missing, further checks won't be possible
 		errs = append(errs, errors.New("--release-repo must be provided"))
-	} else if o.clusterName != "" {
-		buildDir := buildFarmDirFor(o.releaseRepo, o.clusterName)
-		if _, err := os.Stat(buildDir); !os.IsNotExist(err) {
-			errs = append(errs, fmt.Errorf("build farm directory: %s already exists", o.clusterName))
+	} else {
+		if o.createPR {
+			//make sure the release repo is on the master branch and clean
+			if err := os.Chdir(o.releaseRepo); err != nil {
+				errs = append(errs, err)
+			} else {
+				branch, err := exec.Command("git", "branch", "--show-current").Output()
+				if err != nil {
+					errs = append(errs, err)
+				} else if master != strings.TrimSpace(string(branch)) {
```

Comment:
```
git rev-parse --symbolic-full-name --abbrev-ref HEAD
```

Use git plumbing commands when you want machine-readable output.

Reply:
It still needed the trimming, unfortunately.